### PR TITLE
fix postcode address lookup

### PR DESCRIPTION
**Pull request by pvighi**
Co-Authored-By: Tom Richards <twrichards@users.noreply.github.com>

Postcode lookup was only working for the first postcode entered.
While fixing it we noticed that the react component in charge of this didn't need any state so we refactored it.

### CODE REVIEW

File: assets/javascripts/modules/react/addressChooser.jsx [hunk 1]
```diff
@@ -12,55 +14,37 @@ function chooseAddress(addressSection, addressStr) {
...
+
+export default ChooseAddress;
 
-    render() {
-        let createItem = (item, key) => {
-            return <option key={key} value={item.value}>{item.name}</option>;
-        };
-        return (
-            <select className="select select--wide" onChange={this.handleChange} value={this.state.value}>
-                {this.options.map(createItem)}
```
Comment (pvighi) [inline assets/javascripts/modules/react/addressChooser.jsx]: the cause of the bug was that this.options was never updated outside the constructor

Comment (paulbrown1982) [reply assets/javascripts/modules/react/addressChooser.jsx]: Has this bug always been there then?

Comment (pvighi) [reply assets/javascripts/modules/react/addressChooser.jsx]: I don't know when it was introduced but it doesn't seem to be related to the validation bug we fixed yesterday

**Comment by pvighi** [comment]
[Fix home delivery address lookup on subscriptions frontend](https://trello.com/c/NFRaaBxj/194-fix-home-delivery-address-lookup-on-subscriptions-frontend)

**Pull request merged**